src/serde/read_cache_lookup.rs

```diff
@@ -153,7 +154,14 @@ impl ReadCacheLookup {
             let mut new_partial_paths = vec![];
             for (node, path) in partial_paths.iter_mut() {
                 if *node == self.root_hash {
-                    possible_responses.push(reversed_path_to_vec_u8(path));
+                    // make sure we never return a path that needs more (or the
+                    // same) bytes to serialize than the node we're referencing.
+                    if let Some(path_len) = atom_length_bits(path.len() as u64) {
+                        if path_len < max_bytes_for_path_encoding {
```

Comment:
this `<` was a mistake. It should be `<=` since `max_bytes_for_path_encoding` has already subtracted the mandatory 1 byte saving.